File: web/union-sql-inejection-mybatis/blog/src/main/java/com/example/blog/service/PostService.java
package com.example.blog.service;

import com.example.blog.domain.Post;
import com.example.blog.domain.PostSearchParam;
import com.example.blog.mapper.PostMapper;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

import java.util.List;

@RequiredArgsConstructor
@Service
public class PostService {

    private final PostMapper postMapper;

    public List<Post> getPostsBySearchParam(PostSearchParam postSearchParam) {
        return postMapper.selectPostsBySearchParam(postSearchParam);
    }
}

File: web/markdown-xss/blog/src/main/kotlin/org/example/blog/service/PostService.kt
package org.example.blog.service

import org.example.blog.domain.Post
import org.example.blog.repository.PostRepository
import org.springframework.stereotype.Service

@Service
class PostService(private val postRepository: PostRepository) {

    fun getPost(postId: Long): Post {
        return postRepository.findById(postId)
            .orElseThrow {
                IllegalArgumentException("Post not found with ID: $postId")
            }
    }

    fun createPost(post: Post): Post {
        return postRepository.save(post)
    }
}
